api/src/test/java/org/openmrs/api/AdministrationServiceTest.java

```diff
@@ -556,6 +560,208 @@ public void getGlobalProperty_shouldGetPropertyInCaseInsensitiveWay() {
 		assertEquals(orig, noprop);
 	}
 	
+	@Test
+	public void filterGlobalPropertiesByViewPermissions_shouldFilterGlobalPropertiesIfUserIsNotAllowedToViewSomeGlobalProperties() {
+		executeDataSet(ADMIN_INITIAL_DATA_XML);
+		
+		int expectedSize = adminService.getAllGlobalProperties().size();
+		// create a new test global property and add view privileges
+		GlobalProperty property = new GlobalProperty();
+		property.setProperty("test_property");
+		property.setPropertyValue("test_property_value");
+		property.setViewPrivilege(Context.getUserService().getPrivilege("Some Privilege For View Global Properties"));
+		adminService.saveGlobalProperty(property);
+		// assert new test global property is saved properly
+		List<GlobalProperty> properties = adminService.getAllGlobalProperties();
+		assertEquals(expectedSize + 1, properties.size());
+		
+		if (Context.isAuthenticated()) {
```

Comment:
Is it possible to reach here with a non authenticated user?

Reply:
Actually no, removed the redundant checker, thanks for the insight👍🏾